fix: CMB Service not disconnecting

EmandM opened the pull request:
## Purpose of this PR

CMB service doesn't send a UTP disconnection when disconnecting a client after an invalid ConnectionRequest.
This PR ensures that anytime a DisconnectReasonMessage is received when using the CMB service, the NetworkManager is disconnected as expected.

### Jira ticket

[MTTB-1768](https://jira.unity3d.com/browse/MTTB-1768)

### Changelog

- Fixed: Shutdown immediately after a DisconnectReasonMessage is received with CMB service

## Documentation

- No documentation changes or additions were necessary.

## Testing & QA (How your changes can be verified during release Playtest)


### Functional Testing
[//]: # (If checked, List manual tests that have been performed.)
_Manual testing :_
- [ ] `Manual testing done`

_Automated tests:_
- [ ] `Covered by existing automated tests`
- [x] `Covered by new automated tests`

_Does the change require QA team to:_

- [ ] `Review automated tests`?
- [ ] `Execute manual tests`?
- [ ] `Provide feedback about the PR`?

If any boxes above are checked the QA team will be automatically added as a PR reviewer.

## Backports

CMB service is 2.x only

EmandM committed:
fix: CMB Service not disconnecting

EmandM committed:
Update CHANGELOG

NoelStephensUnity committed:
fix

Adding the UnityEngine namespace.

NoelStephensUnity committed:
style

removing extra line.

NoelStephensUnity reviewed (approved):
Looks good to me!
🚀

NoelStephensUnity committed:
style

removing UnityEngine using directive.

NoelStephensUnity committed:
fix

This resolves an issue with the previous fix for the DA service disconnect issue where it should only early exit a disconnect event if there is not connected client and it is the server =or= it is a client connected to the CMB service backend and the clients locally stored transport ID is no longer valid.
